### PR TITLE
Add back files

### PR DESCRIPTION
**Pull request by Matt-Ord**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+/build
+
+# misc
+.DS_Store
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+npm-debug.log*
```
Comment (Matt-Ord) [inline .gitignore:21]: Should also not commit package-lock.json

File: src/Calculator.js [hunk 2]
```diff
@@ -0,0 +1,291 @@
+import { useState, useEffect } from 'react';
+import './Calculator.css';
+
+
+var dpcounter = 0;    /*to ensure no entry of e.g 0.1.1*/
```
Comment (Matt-Ord) [inline src/Calculator.js:5]: Should never use var - use let instead
https://stackoverflow.com/questions/762011/what-is-the-difference-between-let-and-var

File: src/Calculator.js [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+var opcounter = 0;    /*to compute when =2*/
+var anscounter = 0;   /*to stop entry of ans on first use*/
```
Comment (Matt-Ord) [inline src/Calculator.js:7]: All of these variables should be inside the function - it can cause tearing and strange behaviour if two calculators are added to the same page

File: src/Calculator.js [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+    if (document.getElementById('screen').offsetWidth > 
+        document.getElementById('calculator-display').offsetWidth) {  /*stop display overflowing*/
+      setDisp('NaN');
+    }
+  });
+
+  function updateDisp(op) {
+    if (op === 'AC') {
+      setDisp('0');
+      prevop = 0;
```
Comment (Matt-Ord) [inline src/Calculator.js:27]: this should at the very least be 4 functions (one for +-x/, 1234 ect, one for =, one for Ac)

File: src/Calculator.js [hunk 5]
```diff
@@ -0,0 +1,291 @@
+import { useState, useEffect } from 'react';
+import './Calculator.css';
+
+
+var dpcounter = 0;    /*to ensure no entry of e.g 0.1.1*/
+var opcounter = 0;    /*to compute when =2*/
+var anscounter = 0;   /*to stop entry of ans on first use*/
+var prevop = 0;       /*force next to not be an operator e.g stops entering +x*/
+var nextop = 0;       /*force next to be an operator e.g stops entering ans3*/
+var minuscounter = 0; /*stops too many minus signs*/
```
Comment (Matt-Ord) [inline src/Calculator.js:10]: What is minuscounter for?

File: src/Calculator.js [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+            dpcounter = 1;
+          }
+          else {
+            result = disp;
+            anscounter = 1;
+          }
+          break;
+        default:
+          if (nextop === 0) {
+            setDisp(disp+op);
```
Comment (Matt-Ord) [inline src/Calculator.js:182]: I think it would be easier if instead of storing display you store operation, answer and input as three separate values

File: src/setupTests.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+// jest-dom adds custom jest matchers for asserting on DOM nodes.
+// allows you to do things like:
+// expect(element).toHaveTextContent(/react/i)
+// learn more: https://github.com/testing-library/jest-dom
+import '@testing-library/jest-dom';
```
Comment (Matt-Ord) [inline src/setupTests.js:5]: Is this file needed?

File: src/Calculator.css [hunk 8]
```diff
@@ -0,0 +1,61 @@
+#calculator-body {
```
Comment (Matt-Ord) [inline src/Calculator.css:1]: still looks weird if I use devtools to make it an iPhone se

File: src/Calculator.js (hunk 3 above)
Comment (RobertJWatts84) [reply src/Calculator.js:7]: Initially I had the variables inside the function but the trouble was every time setDisp is called it re-renders <Calculator /> which re-initializes the variables all to zero.

File: src/Calculator.js (hunk 2 above)
Comment (RobertJWatts84) [reply src/Calculator.js:5]: Done

File: src/Calculator.js (hunk 4 above)
Comment (RobertJWatts84) [reply src/Calculator.js:27]: This makes far too much sense for me to have not noticed. Oh dear

File: src/Calculator.js (hunk 5 above)
Comment (RobertJWatts84) [reply src/Calculator.js:10]: I wanted to be able to have more than one operator for things like 6x-9 but not to be able to input things like 6x--9 or ----6---4 for example. So minuscounter means you can have one - on either side of the middle operator at the beginning of either number.

File: src/Calculator.js (hunk 3 above)
Comment (Matt-Ord) [reply src/Calculator.js:7]: Should make use of useState in this case

File: src/Calculator.js (hunk 6 above)
Comment (RobertJWatts84) [reply src/Calculator.js:182]: This might make sense if I started that way but I don't think its worth going back through and changing it now?

File: src/setupTests.js (hunk 7 above)
Comment (RobertJWatts84) [reply src/setupTests.js:5]: Its one of the files generated with create-react-app I must admit I didn't go through them all to see what's needed.

File: src/Calculator.js (hunk 6 above)
Comment (Matt-Ord) [reply src/Calculator.js:182]: Yeah up to you at the end of the day but I think it'll be a lot clearer

**Comment by Matt-Ord** [comment]
Should still get rid of the let variables - it's very bad practice

**Comment by Matt-Ord** [comment]
Apart from moving the let variables I still think you could simplify it loads by switching to the method I said before but up to you how much you want to spend making the calculator lol

File: src/Calculator.js [hunk 9]
```diff
@@ -0,0 +1,321 @@
...
+          <div className="calc-button"><button onClick={() => operator("/")}>&#247;</button></div>
+        </div>
+        <div className="row">
+          <div className="calc-button"><button onClick={() => num('7')}>7</button></div>
+          <div className="calc-button"><button onClick={() => num('8')}>8</button></div>
+          <div className="calc-button"><button onClick={() => num('9')}>9</button></div>
+          <div className="calc-button"><button onClick={() => operator("x")}>x</button></div>
+        </div>
+        <div className="row">
+          <div className="calc-button"><button onClick={() => num('4')}>4</button></div>
```
Comment (Matt-Ord) [inline src/Calculator.js:300]: rename functions like num() to onNumber() and operator to onOperator() and pass an int as an argument not a string

File: src/Calculator.js [hunk 10]
```diff
@@ -0,0 +1,321 @@
...
+      else {
+        if (nextop === 0 && zerocounter === 0) {
+          setDisp(disp+inp);
+          prevop = 0;
+        }
+      }
+    }
+  }
+
+  function zero() {
```
Comment (Matt-Ord) [inline src/Calculator.js:199]: should probably merge function with onNumber

File: src/Calculator.js (hunk 9 above)
Comment (RobertJWatts84) [reply src/Calculator.js:300]: I've changed the names but if you pass them as an int then setDisp(disp+inp) just adds the numbers e.g. entering 6 then 3 gives 9 not 63

File: src/Calculator.js (hunk 3 above)
Comment (RobertJWatts84) [reply src/Calculator.js]: That makes sense. Changed

File: src/Calculator.js (hunk 10 above)
Comment (RobertJWatts84) [reply src/Calculator.js:199]: makes it easier to understand the implementation of zerocounter this way imo

File: src/Calculator.js (hunk 9 above)
Comment (Matt-Ord) [reply src/Calculator.js]: In this case do disp + num.toString()

**Comment by Matt-Ord** [comment]
Looks good - there's obviously a load of thing you can do to improve clarity but up to you really. I would get rid of side effects in the isNan function though - I wouldn't expect that to cause a state mutation, and functions like and should also be renamed to onAns
